scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -1909,6 +1910,30 @@ class FormatOps(
     }
   }
 
+  // Optional braces in templates after `:|with`
+  object OptionalBracesTmpl {
+    def unapply(meta: FormatToken.Meta): Option[Template] =
+      meta.leftOwner match {
+        case t: Template if templateCurly(t).contains(tokens(meta.idx).left) =>
+          Some(t)
+        case _ => None
+      }
+  }
+
+  // Optional braces after any token that can start indentation:
+  // ) =  =>  ?=>  <-  catch  do  else  finally  for
+  // if  match  return  then  throw  try  while  yield
+  object OptionalBracesBlck {
+    def unapply(meta: FormatToken.Meta): Option[Term.Block] = {
+      val nft = nextNonComment(tokens(meta.idx))
```

Comment:
allows comments between keyword and the start of the braceless block. also, handles `)` the same way, no need to clarify extension or if.